### PR TITLE
Issue #3214 closure audit: close as completed negative result

### LINKED ISSUE #3214: training: hard-case-focused predictive retraining with crossing-conflict augmentation
## Archetype Metadata

```yaml
archetype: training-campaign
evidence_tier: idea
linked_policy:
  - docs/maintainer_values.md
  - configs/benchmarks/predictive_hard_seeds_v1.yaml
```

## Goal / Problem

The third bet in the hard-case breakthrough portfolio (#C) is **model-side**: the predictive model may
not anticipate crossing-conflict interactions well, so hard cases fail regardless of selection (#3204)
or maneuver authority (sibling issue). This issue tests whether **hard-case-focused retraining / data
augmentation** — oversampling or augmenting crossing-conflict phases in the trajectory-model training
set — raises hard-seed benchmark success.

Runs in parallel with the other two bets; independent so the portfolio can be a true tournament.

## Hypothesis / Claim contract

- **Target hypothesis:** retraining the predictive model with crossing-conflict-weighted data raises
  hard-seed success over the current `predictive_proxy_selected_v1` lineage.
- **Comparator / baseline:** current model lineage on `predictive_hard_seeds_v1`, same algo YAML + seeds.
- **Minimum valid evidence:** ≥ 1 retrained checkpoint vs baseline on the hard-seed benchmark, with the
  navigation gate kept separate from the trajectory gate (`val_ade`/`val_fde`), per existing policy.
- **Decision / stop rule:** adopt if hard success improves with non-regressing overall success; if ADE
  improves but hard success does not, that corroborates the #3204 finding that ADE ≠ benchmark success
  and is recorded as such. Negative result if neither moves.
- **Provenance plan:** training config + data-weighting spec under `configs/`; checkpoint promoted via
  `model/registry.yaml` with W&B run-file metadata (no reliance on `output/tmp`); context note + summary.

## Scope / Out of scope

- In: crossing-conflict-weighted sampling/augmentation in the predictive-model training pipeline; one
  retrained checkpoint; hard-seed evaluation.
- Out: new model architectures (tracked separately, e.g. #2844); paper-grade claim before portfolio
  synthesis (#C).

## Acceptance Criteria

- [ ] Hard-case-weighted training config + data-weighting spec are explicit and reproducible.
- [ ] ≥ 1 retrained checkpoint evaluated vs baseline on `predictive_hard_seeds_v1`; nav gate and ADE
      gate reported separately.
- [ ] Decision per stop rule with uncertainty; checkpoint promoted durably (registry/W&B), not local-only.
- [ ] Result classified on the evidence ladder; no fallback/degraded promotion.

## Validation / Testing

```bash
uv run python scripts/training/train_predictive_model.py --config configs/research/predictive_hardcase_retrain_v1.yaml   # name TBD by implementer
uv run python scripts/validation/evaluate_predictive_planner.py --algo-config <retrained-algo>.yaml --benchmark configs/benchmarks/predictive_hard_seeds_v1.yaml
```

## Workflow

Branch/worktree + online PR only. Suggested: `../robot_sf_ll7.worktrees/issue-NNN-hardcase-retrain`.

## Estimate

Effort: 16–28h + training compute (SLURM). Risk: medium. Resource: SLURM for the training run.

## Blocked by

Needs the hard-case training data path; coordinate with existing trace/data lanes. Portfolio sibling of
#3204; parent = hard-case portfolio epic.

## 🤖 Agent execution spec (codex gpt5.5 medium)

_Appended 2026-06-20. Original body unchanged. Agent runtime: local only. **Config + data-weighting + augmentation code are local**; the retraining run needs SLURM/GPU._

`archetype: training-campaign` · `evidence_tier: blocked` → `nominal` after the retrained checkpoint exists.

### Verified entry points
- Predictive training pipeline: `scripts/training/train_predictive_planner.py`, `scripts/training/run_predictive_training_pipeline.py`.
- Hard-case data: `scripts/training/collect_predictive_hardcase_data.py`, `scripts/training/build_predictive_mixed_dataset.py`.
- Eval: `scripts/validation/evaluate_predictive_planner.py` on `configs/benchmarks/predictive_hard_seeds_v1.yaml`.
- Promotion: `model/registry.yaml` (no `output/tmp` reliance).

### Agent-executable slice (local)
1. Write the crossing-conflict-weighted training config + explicit data-weighting/augmentation spec under `configs/` (oversample/augment crossing-conflict phases in the trajectory-model training set).
2. Implement the augmentation in `build_predictive_mixed_dataset.py` (or a sibling) with a unit test on a small fixture.
3. Keep the navigation gate separate from the trajectory gate (`val_ade`/`val_fde`) per existing policy.

### Acceptance criteria (mirrors body)
- [ ] Hard-case-weighted training config + data-weighting spec explicit + reproducible.
- [ ] ≥1 retrained checkpoint vs baseline on `predictive_hard_seeds_v1` (nav gate ≠ ADE gate).
- [ ] ADE-up-but-hard-success-flat corroborates #3204 and is recorded as such; negative result if neither moves.

### Validation commands
```bash
uv run python scripts/training/build_predictive_mixed_dataset.py --help
uv run python -m pytest tests/ -k "predictive and (dataset or augment or hard)" -q
```

### Blocked-until
- The retrain run is SLURM/GPU (`resource:slurm`). The agent prepares config + augmentation + dataset build + eval harness; SLURM produces the checkpoint.

### PR DESCRIPTION
**Pull request by ll7**
## Reviewer-critical summary

- **What changed:** closure audit for [#3214](https://github.com/ll7/robot_sf_ll7/issues/3214). A new evidence note (`docs/context/evidence/issue_3214_closure_audit_2026-07-06.md`) re-maps the four acceptance criteria to merged evidence and **recommends closing #3214 as a completed negative research result**, superseding the 2026-07-05 `blocked_not_closable` audit.
- **Why now:** #3214 was re-admitted to the ready-queue as a merge-driven closure re-admit (PR #4568 merged, issue stayed open). The model-side hypothesis it asked about has already been *tested and rejected* — this PR resolves the open/closable ambiguity with a decision artifact instead of another blocker restatement.
- **Claim boundary:** closure-audit / research-triage artifact only. No new benchmark, metric, model-provenance, or paper-facing claim; it reclassifies existing tracked evidence. The underlying #3254 result remains an analysis-only negative training result.
- **Required validation:** docs-proof consistency + docs-evidence integrity + bundle checksum (all green, below).
- **Remaining blocker:** none for closure under the negative-result reading. Residual is *maintainer intent* (~80% confidence) — the gate can demote `Closes` if #3214 should stay an umbrella for a future control-law-assisted retry.

### The decision, in short

| Acceptance criterion | Verdict | Evidence |
| --- | --- | --- |
| Hard-case-weighted config + reproducible weighting spec | Met | PR #3255 (mixing spec, `weighting_spec` resolution, oversampling + tests) |
| >=1 retrained checkpoint vs baseline on `predictive_hard_seeds_v1`, nav gate != ADE gate | Met (diagnostic negative) | #3254 / PR #3515 — Slurm 13042: val ADE 0.04837 / FDE 0.09735 vs closed-loop success 0.08696; baselines 0.101 / 0.069 |
| Decision per stop rule + uncertainty; checkpoint promoted durably | Met (negative reading) | Negative decision recorded; W&B run `3tu3tmee` + SHA256-checked bundle; registry promotion of a gate-failing checkpoint correctly withheld per fail-closed policy |
| Result classified; no fallback/degraded promotion | Met | Classified `negative_training_result_not_benchmark_promotion` |

The prior audit held the issue open on two mis-readings, both corrected in the note: (1) "checkpoint promoted durably" — durable provenance exists via W&B + bundle; promoting a gate-failing checkpoint is *prohibited* by the fail-closed policy, so it is not an unmet criterion; (2) "public control-law-side change" is a **distinct follow-up bet**, out of #3214's own stated model-side scope (*"Out: new model architectures (tracked separately)"*). Sequencing predecessor #3216 is now closed.

## Contract delta

- No schema, code, or runtime-behavior change. No new fail-closed blocker.
- `docs/context/catalog.yaml`: prior note marked `status: superseded` + `replacement`; new note registered `status: evidence`.
- The tracked readiness packet `configs/training/predictive/predictive_retraining_readiness_issue_3214.yaml` is intentionally **unchanged**: its `launch_decision.state: blocked_until_control_law_change` governs whether to *re-launch* a run, not whether #3214 is complete. The note explains why the launch-block and issue-closure are consistent.

## Validation

```
sha256sum -c (issue_3254 bundle)                        -> README.md: OK, summary.json: OK
check_docs_proof_consistency.py (new note + bundle)     -> OK, exit 0
check_docs_evidence_integrity.py --base-ref origin/main -> 4 changed file(s) passed, exit 0
catalog.yaml YAML parse                                 -> OK (827 entries)
quoted metrics vs summary.json                          -> exact match (13042 / 3tu3tmee / 0.04837 / 0.09735 / 0.08696 / gate 0.30)
```

## Closure

Closes #3214

## Out of scope (explicit)

- No full benchmark campaign run, no Slurm/GPU submission, no compute of any kind.
- No paper/dissertation claim edits; no model checkpoint promotion.
- No new retraining tooling/guard (would duplicate PR #3255 / #4467 and is held by the readiness packet).

<details>
<summary>Governance / propagation</summary>

- **Downstream propagation:** single durable state surface updated = the closure-audit note + its catalog/README registration; no separate issue comment added (worker is not authorized to comment, and the note is the canonical surface). Parent epic #3215 already tracks the portfolio outcome.
- **Research Result Guidance:** target hypothesis = model-side crossing-conflict retraining raises hard-seed success; comparator = current lineage on `predictive_hard_seeds_v1`; evidence tier = diagnostic negative (analysis-only); result classification = negative, corroborates #3204 (ADE != benchmark success); stop rule = negative-result branch fired; synthesis surface = #3215 portfolio.
- **Fragmentation guard:** this is a closure decision, not a micro-guard/packet-refresh; <=1 audit PR merged in the prior 24h.
- **If the gate disagrees with closure:** demote `Closes` to keep #3214 open and open a dedicated control-law-side retraining issue rather than reopening this model-side bet.
</details>

### CODE REVIEW

**Comment by ll7** [comment]
## Gate review — findings (phase 1 + phase 2)

**Scope:** docs-only closure audit for #3214 (+121/-6): adds `issue_3214_closure_audit_2026-07-06.md`, marks the 2026-07-05 note `superseded` in `catalog.yaml`, updates the evidence README. No CHANGELOG. No behavior change.

### Phase 1 — verdict on the closure argument (gate ratification)
This supersedes the prior `blocked_not_closable` audit with an honest **negative-result** reading, and I **ratify** it:
- The model-side bet (#3214) was tested via #3254 (Slurm `13042`): trajectory gate passed (ADE 0.04837 / FDE 0.09735) but closed-loop success **0.08696 < 0.30 gate** → rejected. This is the issue's own stop-rule branch ("ADE improves but hard success does not → corroborates #3204"), so it's a legitimate recorded negative result, not an unmet criterion.
- The two items the prior audit held open are correctly re-read: (a) registry promotion of a **gate-failing** checkpoint is *prohibited* by fail-closed policy — withholding it is compliance, not incompleteness (durable provenance still exists via W&B `3tu3tmee` + SHA256 bundle); (b) the control-law-side change is out of #3214's stated model-side scope ("Out: new model architectures") and belongs to a separate follow-up. The maintainer's one sequencing constraint ("after #3216") is satisfied — **#3216 is CLOSED**.
- `Closes #3214` is defensible at ~80% stated confidence with an explicit demote-if-disagree escape hatch. The readiness packet's `launch_decision.state: blocked_until_control_law_change` governs *re-launch*, not issue completeness — correctly reasoned.

### Open review threads (exhaustive): 1 — gemini-code-assist MEDIUM — **FIXED** in phase 2
1. **`issue_3214_closure_audit_2026-07-06.md:58`** — the audit cited `retrain_result.status` but the actual YAML key in `predictive_retraining_readiness_issue_3214.yaml` is **`prior_result.status`** (verified: line 44-45). A factual error in an evidence doc — **FIXED** (not waived, since it's a correctness/accuracy finding).

**Verdict:** honest negative-result closure, well-argued, fail-closed-consistent. Merging once CI on head `8fdab75` is green.
